src/Simplex/FileTransfer/Agent.hs

```diff
@@ -179,3 +182,66 @@ runXFTPLocalWorker c@AgentClient {subQ} doWork = do
                 pure $ s <> chunk
             )
             LB.empty
+
+sendFile :: AgentMonad m => AgentClient -> UserId -> FilePath -> FilePath -> m Int64
+sendFile c userId xftpPath filePath = do
+  -- db: create file in status New without chunks
+  -- add local snd worker for encryption
+  -- return file id to client
+  undefined
+
+runXFTPSndLocalWorker :: forall m. AgentMonad m => AgentClient -> TMVar () -> m ()
+runXFTPSndLocalWorker c@AgentClient {subQ} doWork = do
+  forever $ do
+    void . atomically $ readTMVar doWork
+    runXftpOperation
+  where
+    runXftpOperation :: m ()
+    runXftpOperation = do
+      -- db: get next snd file to encrypt (in status New)
+      -- ? (or Encrypted to retry create? - see below)
+      -- with fixed retries (?) encryptFile
+      undefined
+    encryptFile :: SndFile -> m ()
+    encryptFile sndFile = do
+      -- if enc path exists, remove it
+      -- if enc path doesn't exist:
+      --   - choose enc path
+      --   - touch file, db: update enc path (?)
+      -- calculate chunk sizes, encrypt file to enc path
+      -- calculate digest
+      -- prepare chunk specs
+      -- db:
+      --   - update file status to Encrypted
+      --   - create chunks according to chunk specs
+      -- ? since which servers are online is unknown,
+      -- ? we can't blindly assign servers to replicas.
+      -- ? should we XFTP create chunks on servers here,
+      -- ? with retrying for different servers,
+      -- ? keeping a list of servers that were tried?
```

Comment:
yes, that's what we do when we create connection - similar logic can be used here

Reply:
I don't think it's necessary to keep tried servers in db though, it'll be just during the session

Reply:
yes, the intention was to keep servers list in memory, could even be for chunk retries